Src/Support/Google.Apis.Auth/OAuth2/IAcceptsQuotaProject.cs

```diff
@@ -0,0 +1,33 @@
+﻿/*
+Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+namespace Google.Apis.Auth.OAuth2
+{
+    /// <summary>
+    /// Credential that accepts a quota project to be set.
+    /// Usually, if a quota project is set, it will be send along
+    /// the access token as an extra header when using the credential for
+    /// accessing Google APIs.
+    /// </summary>
+    public interface IAcceptsQuotaProject : ICredential, ITokenAccessWithHeaders
+    {
+        /// <summary>
+        /// The ID of the project associated to this credential for the purposes of
+        /// quota calculation and billing. May be null.
+        /// </summary>
+        string QuotaProject { get; set; }
```

Comment:
I'm nervous about making this mutable within the interface. I'd rather we followed the "with" convention we have elsewhere.

Reply:
We have chatted about this offline and will keep credentials inmutable.